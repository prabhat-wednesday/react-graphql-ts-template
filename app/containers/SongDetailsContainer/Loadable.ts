import loadable from '@app/utils/loadable';

export default loadable(() => import('./index'));
