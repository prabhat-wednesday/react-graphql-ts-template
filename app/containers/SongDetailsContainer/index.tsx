import React from 'react';
import { compose } from '@reduxjs/toolkit';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { useParams } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { selectSongData } from '../ItunesContainer/selector';
import { Song } from '../ItunesContainer/types';
import { Card } from 'antd';
import { get } from 'lodash-es';
import styled from 'styled-components';
import { media } from '@app/themes';

const { Meta } = Card;

interface SongDetailsProps {
  songData: {
    results: Song[];
  };
}

const CustomCard = styled(Card)`
  && {
    padding: 5px;
    margin: 0.5rem;
    width: 17rem;

    ${media.lessThan('tablet')`
      width: 12rem;
      margin: 0.5rem;
    `}
  }
`;

const StyledImg = styled.img`
  && {
    max-height: 15rem;
    object-fit: cover;
  }
`;

const StyledMeta = styled(Meta)`
  && {
    padding: 5px;
  }
`;

const CustomSongDetailsContainer = styled.div`
  display: flex;
  justify-content: center;
`;

const SongDetails = ({ songData }: SongDetailsProps) => {
  const results = get(songData, 'results', []);
  const params = useParams<{ trackId?: string }>();
  const { trackId } = params;
  const singleSong = results?.find((result) => result.trackId === Number(trackId));
  const { artistName, collectionName, artworkUrl100 } = singleSong ?? {};

  return (
    <CustomSongDetailsContainer>
      <Helmet>
        <title>Song Details</title>
        <meta name="description" content="Description of SongDetails" />
      </Helmet>
      <CustomCard
        hoverable
        bodyStyle={{ padding: '10px' }}
        cover={<StyledImg src={artworkUrl100} loading="lazy" data-testid="cover-img" />}
      >
        <StyledMeta title={artistName} description={collectionName} />
      </CustomCard>
    </CustomSongDetailsContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  songData: selectSongData()
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect)(SongDetails);
