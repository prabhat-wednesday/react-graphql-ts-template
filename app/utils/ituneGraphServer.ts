import ApolloClient, { DocumentNode, InMemoryCache } from 'apollo-boost';

export const client = new ApolloClient({
  uri: 'https://itunes.apple.com/search?term=',
  cache: new InMemoryCache()
});

export interface GqlQueryReponse<Data> {
  data?: Data;
  error?: any;
  ok: boolean;
}

export const getQueryResponse = <Data, Variables>(
  query: DocumentNode,
  variables?: Variables
): Promise<GqlQueryReponse<Data>> => {
  console.log(query, 'in graphql server');
  return client
    .query<Data, Variables>({ query, variables })
    .then((res) => {
      console.log(res, 'in graphql');
      if (res.errors) {
        console.log(res);
        return { error: res.errors, ok: false };
      } else {
        console.log(res);
        return { data: res.data, ok: true };
      }
    })
    .catch((err) => {
      return { error: err, ok: false };
    });
};
