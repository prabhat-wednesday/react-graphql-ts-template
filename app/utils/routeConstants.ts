export type RouteConstant = {
  route: string;
  exact?: boolean;
  isProtected?: boolean;
  props?: object;
};

const routeConstants: Record<string, RouteConstant> = {
  itune: {
    route: '/',
    exact: true
  },
  song: {
    route: '/song/:trackId',
    exact: true
  }
};

export type RouteKeys = keyof typeof routeConstants;

export default routeConstants;
