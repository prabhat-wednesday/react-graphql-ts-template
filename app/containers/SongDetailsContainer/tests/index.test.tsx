import React from 'react';
import { renderProvider } from '@app/utils/testUtils';
import SongDetailsContainer from '..';

describe('<SongDetailsContainer test', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<SongDetailsContainer />);
    expect(baseElement).toMatchSnapshot();
  });
});
