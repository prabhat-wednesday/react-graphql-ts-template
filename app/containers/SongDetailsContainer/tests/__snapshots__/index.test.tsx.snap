// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongDetailsContainer test should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="SongDetailsContainer__CustomSongDetailsContainer-gu3src-4 cTRwQg"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable SongDetailsContainer__CustomCard-gu3src-0 lkidkL"
      >
        <div
          class="ant-card-cover"
        >
          <img
            class="SongDetailsContainer__StyledImg-gu3src-1 hnNURB"
            data-testid="cover-img"
            loading="lazy"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta SongDetailsContainer__StyledMeta-gu3src-2 bKPOYA"
          />
          <p
            class="T__StyledText-gcrjip-0 gIxZML SongDetailsContainer__StyledT-gu3src-3 eBnitJ"
            data-testid="t"
          >
            $ undefined
          </p>
          <p
            class="T__StyledText-gcrjip-0 gIxZML SongDetailsContainer__StyledT-gu3src-3 eBnitJ"
            data-testid="t"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
